jgrapht-core/src/main/java/org/jgrapht/GraphTests.java

```diff
@@ -479,6 +479,34 @@
         return new HierholzerEulerianCycle<V, E>().isEulerian(graph);
     }
 
+    /**
+     * Tests whether an undirected graph meets Ore's condition to be Hamiltonian.
+     *
+     * Let G be a (finite and simple) graph with n ≥ 3 vertices. We denote by deg v the degree of a vertex v in G, i.e. the number of incident edges in G to v.
+     * Then, Ore's theorem states that if deg v + deg w ≥ n for every pair of distinct non-adjacent vertices v and w of G, then G is Hamiltonian.
+     *
+     * @param graph the input graph
+     * @param <V> the graph vertex type
+     * @param <E> the graph edge type
+     * @return true if the graph meets Ore's condition, false otherwise
+     * @see org.jgrapht.alg.tour.PalmerHamiltonianCycle
+     */
+    public static <V, E> boolean hasOreProperty(Graph<V, E> graph){
+        Objects.requireNonNull(graph, GRAPH_CANNOT_BE_NULL);
+
+        final int n = graph.vertexSet().size();
+
+        if (!graph.getType().isSimple() || n < 3)
```

Comment:
Now I understand what you were saying. I missed this as it was not at the very beginning of the method.

Reply:
No problem, now the docs are even better!